### PR TITLE
Use lazy load for pages + optimize hooks and break big components into smaller

### PR DESCRIPTION
**Pull request by T-Damer**
Was
image

Now
image

### CODE REVIEW

File: src/hooks/useThrottle.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export default function (callback: (...args: unknown[]) => void, delay = 20) {
+  const isThrottled = useRef(false)
+
+  const throttledCallback = useCallback(
+    (...args) => {
+      if (isThrottled.current) return
+
+      callback(args)
+      isThrottled.current = true
+      setTimeout(() => (isThrottled.current = false), delay)
```
Comment (backmeupplz) [inline src/hooks/useThrottle.ts:12]: Use `{}`

File: src/hooks/useThrottle.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useCallback, useRef } from 'react'
+
+export default function (callback: (...args: unknown[]) => void, delay = 20) {
```
Comment (backmeupplz) [inline src/hooks/useThrottle.ts:3]: Let's do `delayMs`

File: src/components/LazyComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Suspense } from 'react'
+import Loading from 'icons/Loading'
+
+export default function ({ lazyImported }: { lazyImported: JSX.Element }) {
+  return <Suspense fallback={<Loading screenCentre />}>{lazyImported}</Suspense>
```
Comment (backmeupplz) [inline src/components/LazyComponent.tsx:5]: Can you show a few screenshots of how it looks like, please?

**Review by backmeupplz — CHANGES_REQUESTED** [review]

File: src/components/LazyComponent.tsx (hunk 3 above)
Comment (T-Damer) [reply src/components/LazyComponent.tsx:5]: https://user-images.githubusercontent.com/49658988/169561990-67cb2d7f-0d48-4831-80d9-190f359aaaa1.mp4

**Review by backmeupplz — APPROVED** [review]

**Review by backmeupplz — APPROVED** [review]

**Pull request merged**